enclave/chain_relay/src/lib.rs

```diff
@@ -56,18 +56,170 @@ pub type Digest = DigestG<<BlakeTwo256 as HashT>::Output>;
 
 pub type AuthorityListRef<'a> = &'a [(AuthorityId, AuthorityWeight)];
 
+pub trait Validator: Encode + Decode + Clone + Default {
```

Comment:
Introduced new trait for the `LightValidation` struct, to have an additional layer of abstraction